### ISSUE TITLE
How to achieve reasoning on large-scale graph-level tasks

### ISSUE THREAD

**Issue by Jedges** [JHL-HUST/NAGphormer#9]
Your work is simple and effective. I would like to ask if you have considered how this design pattern can be used to  train on graph-level tasks. In this case, a significant problem is that the loss function I set for the graph cannot be calculated within the batch divided by nodes. For small-scale graph data, I can adjust and put the nodes of a graph into one batch, but when the scale of the graph is large, this is not realistic. I would be grateful for your reply.

**Comment by guangnianchenai** [comment]
Thank you for your attention! Actually, I haven't considered applying NAGphormer to graph-level tasks yet. Since graph-level tasks typically don't encounter out-of-memory (OOM) issues, the native self-attention mechanism can be used directly for modeling. Maybe you can consider use 3-order tensor to store intput data [b, n, k], where b is the batchsize (the number of graphs), n is the number of nodes (maybe need padding operation), k is the length of the token sequence.

**Comment by Jedges** [comment]
Thanks for your reply!  That's a good suggestion, I will try it.

**Comment by guangnianchenai** [comment]
Wishing you a successful experiment in advance.
